lwt/protocol_lwt.ml

```diff
@@ -47,7 +47,9 @@ module M = struct
           | Unix.Unix_error((Unix.ECONNREFUSED | Unix.ECONNABORTED | Unix.ENOENT), _, _) ->
             Lwt_unix.sleep 5. >>= fun () ->
             loop ()
-          | e -> fail e
+          | e ->
+            Lwt_unix.close fd;
```

Comment:
If the `fd` has been closed at line 43 this will fail with `Unix.EBADF` and shadow the real exception below. You should probably put it in a `Lwt.catch` ignoring any error in this line

Reply:
Also don't use the semicolon but bind it to the Lwt.fail below

Reply:
Updated.